### PR TITLE
add a search in the "customize toolbar" dialog

### PR DESCRIPTION
**Pull request by peterremote1980**
add a search in the "customize toolbar" dialog

### CODE REVIEW

**Comment by juneau001** [comment]
Thanks for the PR.  It looks like the license headers were removed.  I believe that those should be added back before this PR could be approved.  Thanks again

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.form [hunk 1]
```diff
@@ -1,26 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 
-<!--
```
Comment (emilianbold) [inline core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.form]: Re-add the license header.

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java [hunk 2]
```diff
@@ -107,6 +107,7 @@ public void run() {
         validate();
         repaint();
         setCursor( Cursor.getDefaultCursor() );
+		explorerManager = new ExplorerManager();
```
Comment (emilianbold) [inline core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java]: I wonder why this was moved here.

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java [hunk 3]
```diff
@@ -176,7 +177,7 @@ public void run() {
      * WARNING: Do NOT modify this code. The content of this method is
      * always regenerated by the Form Editor.
      */
-    // <editor-fold defaultstate="collapsed" desc=" Generated Code ">//GEN-BEGIN:initComponents
+    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
```
Comment (emilianbold) [inline core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java]: How could this line change?

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java [hunk 4]
```diff
@@ -329,6 +346,23 @@ private void switchIconSize(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_s
               tbConf.refresh();
           }
     }//GEN-LAST:event_switchIconSize
+
+    private void searchTextFieldKeyReleased(java.awt.event.KeyEvent evt) {//GEN-FIRST:event_searchTextFieldKeyReleased
+        System.out.println(searchTextField.getText());
```
Comment (emilianbold) [inline core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java]: Remove System.out.

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java [hunk 5]
```diff
@@ -502,6 +542,9 @@ private boolean doAcceptDataObject(DataObject obj) {
                     Object instance = instanceCookie.instanceCreate();
                     if( null != instance ) {
                         if( instance instanceof Action ) {
+						   if (!obj.getName().toLowerCase().contains(searchTextField.getText().toLowerCase())){
```
Comment (emilianbold) [inline core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java]: Hm, I wonder if this gets called in the Event Queue?

**Comment by emilianbold** [comment]
> so should i close this and re-submit the PR?

No, you can just (force) push the changes to your peterremote1980 repository and the PR will automatically update itself.

File: core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java (hunk 2 above)
Comment (emilianbold) [reply core.windows/src/org/netbeans/core/windows/view/ui/toolbars/ConfigureToolbarPanel.java]: But why? The field was initialised directly. Does it matter to initialise it here?

**Comment by emilianbold** [comment]
PS: I see GitHub complaining that there are some 'Conflicting files'. So, make sure you also rebase your code perhaps? Usually conflicts mean the code changed in the meantime.

**Comment by matthiasblaesing** [comment]
I have two formal replies:
- this changeset is based on a _very_ old version (hence the conflicts) - the base of the changeset is from May of last year. Please rebase the changeset onto current master.
- The author information is imcomplete and inconsistent. The first commit refers to an @quantr.hk adress, combined with only the firstname as author name and the second commit uses a @gmail.com adress without any realname. Please make sure, that you use a valid Emailadress and complete author name (First+Lastname)

**Comment by peterremote1980** [comment]
> PS: I see GitHub complaining that there are some 'Conflicting files'. So, make sure you also rebase your code perhaps? Usually conflicts mean the code changed in the meantime.

the code changes is based on the master branch of the github source, did i do it wrongly?

**Comment by peterremote1980** [comment]
> I have two formal replies:
> 
> * this changeset is based on a _very_ old version (hence the conflicts) - the base of the changeset is from May of last year. Please rebase the changeset onto current master.
> * The author information is imcomplete and inconsistent. The first commit refers to an @quantr.hk adress, combined with only the firstname as author name and the second commit uses a @gmail.com adress without any realname. Please make sure, that you use a valid Emailadress and complete author name (First+Lastname)

Thanks, for the first point, i was modifying the code based on the master branch. If it is too old, where is the latest code i should start working on?

**Comment by gilbertoca** [comment]
> This branch is 2 commits ahead, 969 commits behind apache:master. 

Your fork is very old.

**Comment by peterremote1980** [comment]
> > PS: I see GitHub complaining that there are some 'Conflicting files'. So, make sure you also rebase your code perhaps? Usually conflicts mean the code changed in the meantime.
> 
> the code changes is based on the master branch of the github source, did i do it wrongly?

i don't understand why, i just forked it 3 days ago.

**Comment by matthiasblaesing** [comment]
How did you "fork"? You need to remember, that git is distributed and if you for (checkout/create a branch), you are branching your current state. If you cloned your repository a few months ago, that would explain it. The repository needs to be kept in sync with upstream (the apache/github copy of the repository). But if this is unclear, please ask on the mailing list, a PR is not the place to discuss this.

**Comment by peterremote1980** [comment]
> How did you "fork"? You need to remember, that git is distributed and if you for (checkout/create a branch), you are branching your current state. If you cloned your repository a few months ago, that would explain it. The repository needs to be kept in sync with upstream (the apache/github copy of the repository). But if this is unclear, please ask on the mailing list, a PR is not the place to discuss this.

Thanks i just click the fork button on github, then git clone it. I think my steps are right.

**Comment by emilianbold** [comment]
No, something odd with your repository, it's not up to date. See:

remotevechie

**Comment by matthiasblaesing** [comment]
No update after months - closing.

**Pull request closed without merging**